File: cordapp/src/main/java/net/corda/derivativestradingnetwork/TokenTransferFlowResponder.java
package net.corda.derivativestradingnetwork;

import co.paralleluniverse.fibers.Suspendable;
import net.corda.core.flows.*;
import net.corda.core.transactions.SignedTransaction;
import net.corda.core.utilities.ProgressTracker;

@InitiatedBy(TokenTransferFlow.class)
public class TokenTransferFlowResponder extends FlowLogic<SignedTransaction> {

    private final FlowSession otherPartyFlow;

    public TokenTransferFlowResponder(FlowSession otherPartyFlow) {
        this.otherPartyFlow = otherPartyFlow;
    }

    @Suspendable
    @Override
    public SignedTransaction call() throws FlowException {
        class SignTxFlow extends SignTransactionFlow {
            private SignTxFlow(FlowSession otherPartyFlow, ProgressTracker progressTracker) {
                super(otherPartyFlow, progressTracker);
            }

            @Override
            protected void checkTransaction(SignedTransaction stx) {

            }
        }

        return subFlow(new SignTxFlow(otherPartyFlow, SignTransactionFlow.Companion.tracker()));
    }
}

File: cordapp/src/main/kotlin/net/corda/derivativestradingnetwork/flow/TokenTransferFlowResponder.kt
package net.corda.derivativestradingnetwork.flow

import co.paralleluniverse.fibers.Suspendable
import net.corda.core.flows.*
import net.corda.core.transactions.SignedTransaction
import net.corda.core.utilities.ProgressTracker

@InitiatedBy(TokenTransferFlow::class)
class TokenTransferFlowResponder(private val otherPartyFlow: FlowSession) : FlowLogic<SignedTransaction>() {

    @Suspendable
    @Throws(FlowException::class)
    override fun call(): SignedTransaction {
        class SignTxFlow constructor(otherPartyFlow: FlowSession, progressTracker: ProgressTracker) : SignTransactionFlow(otherPartyFlow, progressTracker) {

            override fun checkTransaction(stx: SignedTransaction) {

            }
        }

        return subFlow(SignTxFlow(otherPartyFlow, SignTransactionFlow.tracker()))
    }
}